### Waypoints ###
### Controller Parameters

lqr_controller: {

hover_throttle: 0.5,
drag_constant: 0.1,

lqr_max_pos_error: 0.2,
lqr_max_alt_error: 0.1,
lqr_max_vel_error: 5.,
lqr_max_ang_error: 0.2,
#lqr_max_pos_error: .5,
#lqr_max_vel_error: 1.,
#lqr_max_ang_error: .3,

lqr_max_throttle_command: 1.,
lqr_min_throttle_command: 0.,
lqr_max_omega_command: 2.,
lqr_min_omega_command: -2.,

# Gains for dxPos, dxVel, dxAtt
#Q: [2., 2., 2., 1., 1., 1., 3.33, 3.33, 3.33],
Q: [10., 10., 20., 10., 10., 10., 1., 1., 1.],
#Q: [100., 100., 100., 1., 1., 1., 10., 10., 10.],

# Gains for uThrottle, uOmega
#R: [5., 1.5, 2., 2.],
R: [10., 2., 2., 1.0],

# Trajectory
# Fig8
figure8_trajectory: true,
fig8_x_width: 1.5,
fig8_y_width: 1.0,
fig8_z_width: 0.,
fig8_period: 50.,

fig8_x0: [0., 0., -3.,          # POS
          0., 0., 0.,           # VEL
          1., 0., 0., 0.,       # ATT
          0., 0., 0.],          # OMEGA

# Waypoints
waypoint_trajectory: false,
waypoint_duration: 5.,
#waypoints: [0., 0., -5., 0.,    # N, E, D, YAW
            #0., -5., -5., 1.,
            #-5., 0., -5., 2.,
            #0., 0., -1., 3.,
            #0., 0., -10., 0.],
waypoints: [0., 0., -5., 0.,    # N, E, D, YAW
            2., 0., -5., 0.,
            0., 2., -5., 0.,
            -2., 0., -5., 0.,
            0., -2., -5., 0.],
}
